clients/client/src/main/java/org/projectnessie/client/http/HttpResponse.java

```diff
@@ -44,13 +46,34 @@ private <V> V readEntity(ObjectReader reader) {
         return null;
       }
       try (InputStream is = responseContext.getInputStream()) {
+        if (!responseContext.isJsonCompatibleResponse()) {
+          nonJsonResponse();
+        }
+
         return reader.readValue(is);
       }
     } catch (IOException e) {
       throw new HttpClientException("Cannot parse request.", e);
     }
   }
 
+  private void nonJsonResponse() throws IOException {
+    Status status = responseContext.getResponseCode();
+    throw new NessieServiceException(
```

Comment:
I wonder if a `NessieBadResponseException` (extends `NessieServiceException`) may be more appropriate :thinking: It would allow interested clients to process this error case in a more robust way. WDYT?